lib/redlock/client.rb

```diff
@@ -262,8 +279,15 @@ def lock_instances(resource, ttl, options)
       value = (options[:extend] || { value: SecureRandom.uuid })[:value]
       allow_new_lock = options[:extend_only_if_locked] ? 'no' : 'yes'
 
+      errors = []
+
       locked, time_elapsed = timed do
-        @servers.select { |s| s.lock resource, value, ttl, allow_new_lock }.size
+        @servers.select do |s|
+          s.lock(resource, value, ttl, allow_new_lock)
+        rescue => e
+          errors << e
+          false
+        end.size
```

Comment:
instead of `servers.select { ... }.size` you could just have `servers.count { ... }` by leveraging [`Array#count`](https://ruby-doc.org/core-3.0.0/Array.html#method-i-count):

> With no argument and a block given, calls the block with each element; returns the count of elements for which the block returns a truthy value:
> 
> ```ruby
> [0, 1, 2, 3].count {|element| element > 1} # => 2
> ```